### PR TITLE
feat(rpc) Implement `Filecoin.EthCall`

### LINKED ISSUE #4613: `EthCall`

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:

- implement RPC method `Filecoin.EthCall`
- API compare test

```console
$ ./forest-tool api compare --lotus /ip4/127.0.0.1/tcp/1234/http --forest /ip4/127.0.0.1/tcp/2345/http --filter EthCall
| RPC Method       | Forest | Lotus |
|------------------|--------|-------|
| Filecoin.EthCall | Valid  | Valid |
```

## Reference issue to close (if applicable)

Closes #4613

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/rpc/methods/eth.rs [hunk 1]
```diff
@@ -1870,6 +1870,36 @@ impl RpcMethod<1> for EthGetTransactionHashByCid {
...
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (tx, block_param): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let msg = tx.try_into()?;
+        let ts = tipset_by_block_number_or_hash(ctx.chain_store(), block_param)?;
+        let invoke_result = ctx.state_manager.call(&msg, Some(ts))?;
+
+        if msg.to() == FilecoinAddress::ETHEREUM_ACCOUNT_MANAGER_ACTOR {
+            // As far as I can tell, the Eth API always returns empty on contract deployment
```
Comment (LesnyRumcajs) [inline src/rpc/methods/eth.rs:1892]: Let's have it less subjective and support it with some helpful links.

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
@elmattic can we now improve the error messages in other eth methods? I recall you mentioning that some time ago that this method was a blocker for this.

**Review by hanabi1224 — APPROVED** [review]

File: src/rpc/methods/eth.rs (hunk 1 above)
Comment (elmattic) [reply src/rpc/methods/eth.rs:1892]: I will remove that comment (it came from the Lotus codebase). I can't find a proper Ethereum RPC specification that I could link to, sadly.

**Comment by elmattic** [comment]
> @elmattic can we now improve the error messages in other eth methods? I recall you mentioning that some time ago that this method was a blocker for this.

I'm sorry, but I don't remember. If this rings a bell, I will open an issue.

**Comment by LesnyRumcajs** [comment]
> > @elmattic can we now improve the error messages in other eth methods? I recall you mentioning that some time ago that this method was a blocker for this.
> 
> I'm sorry, but I don't remember. If this rings a bell, I will open an issue.

It's this, no? https://github.com/ChainSafe/forest/issues/4554

**Comment by elmattic** [comment]
> > > @elmattic can we now improve the error messages in other eth methods? I recall you mentioning that some time ago that this method was a blocker for this.
> > 
> > 
> > I'm sorry, but I don't remember. If this rings a bell, I will open an issue.
> 
> It's this, no? #4554

Indeed!

**Pull request merged**